### PR TITLE
Fix unmute not restoring audible volume when stored volume is 0

### PR DESCRIPTION
**Pull request by dweinber**
## Description
In some cases the player could not be unmuted anymore via the unmute button, only with the volume slider. This is bad UX, but even worse, on mobile devices, we don't show the slider, which means the user can never unmute in that scenario.

- Prevent `VolumeController` from storing a volume to restore of `0`, so that unmuting always restores an audible level
- When recalling volume, always unmute and default to `100` if the stored volume is `0`
- Add tests covering the muted and zero-volume edge cases

## Checklist (for PR submitter and reviewers)

- [x] `CHANGELOG` entry

### CODE REVIEW

File: src/ts/utils/VolumeController.ts [hunk 1]
```diff
@@ -87,7 +89,9 @@ export class VolumeController {
     const uiMuted = playerMuted || playerVolume === 0;
     const uiVolume = playerMuted ? 0 : playerVolume;
 
-    this.storeVolume();
+    if (!playerMuted && playerVolume > 0) {
+      this.storeVolume();
+    }
```
[the remark this answers is not in the document]
Comment (dweinber) [reply src/ts/utils/VolumeController.ts]: Fixed in 5795384d

File: spec/utils/VolumeController.spec.ts [hunk 2]
```diff
@@ -24,5 +26,35 @@ describe('VolumeController', () => {
 
       expect(volumeController.storeVolume).toHaveBeenCalledTimes(1);
     });
+
+    it('should not update the stored volume when player is muted', () => {
+      (playerMock.isMuted as jest.Mock).mockReturnValue(true);
+      (playerMock.getVolume as jest.Mock).mockReturnValue(0);
+      volumeController.storeVolume = jest.fn();
```
[the remark this answers is not in the document]
Comment (dweinber) [reply spec/utils/VolumeController.spec.ts]: Covered in 8c968745

File: spec/utils/VolumeController.spec.ts [hunk 3]
```diff
@@ -24,5 +26,58 @@ describe('VolumeController', () => {
...
+      // storeVolume should not persist zero — verify via the private field
+      expect((volumeController as any).storedVolume).not.toBe(0);
+    });
+  });
+
+  describe('recallVolume', () => {
+    it('should default to volume 100 when stored volume is 0', () => {
+      // Constructor calls storeVolume() which gets 0 — should not persist it
+      (playerMock.getVolume as jest.Mock).mockReturnValue(0);
+      volumeController = new VolumeController(playerMock);
```
[the remark this answers is not in the document]
Comment (dweinber) [reply spec/utils/VolumeController.spec.ts]: Fixed in 3c85897e and bc6dd79b

File: CHANGELOG.md [hunk 4]
```diff
@@ -11,6 +11,10 @@ and this project adheres to [Semantic Versioning](http://semver.org/).
 
 - `AdCounterLabel` now shows the ad position across multiple ad breaks scheduled at the same time (e.g. `Ad 2 of 3` instead of `Ad 1 of 1` for each)
 
+### Fixed
+
+- `VolumeController` can store a volume to restore of `0` in some cases, causing unmute to not restore an audible volume level
```
[the remark this answers is not in the document]
Comment (dweinber) [reply CHANGELOG.md:16]: This is the form we use, so using this as false positive

**Review by stonko1994 — COMMENTED** [review]
In common use cases, it works great. However, there is one use-case where it still doesn't work nicely.

When you scrub the volume slider slowly enough the latest value stored will be `1`. This value will be restored after unmuting.

Can we adapt the volume controller to not store values during scrubbing, but only store after scrubbing is finished?

File: spec/utils/VolumeController.spec.ts [hunk 5]
```diff
@@ -17,12 +17,144 @@ describe('VolumeController', () => {
...
         timestamp: Date.now(),
       });
 
       expect(volumeController.storeVolume).toHaveBeenCalledTimes(1);
     });
+
+    it('should not update the stored volume when player is muted with zero volume', () => {
+      (playerMock.getVolume as jest.Mock).mockReturnValue(50);
+      volumeController = new VolumeController(playerMock);
+
```
[the remark this answers is not in the document]
Comment (dweinber) [reply spec/utils/VolumeController.spec.ts:31]: Fixed in a40eb7b9

File: src/ts/utils/VolumeController.ts [hunk 6]
```diff
@@ -114,6 +132,8 @@ export class VolumeTransition {
   }
 
   finish(volume: number): void {
+    this.controller.endTransition();
+
```
[the remark this answers is not in the document]
Comment (dweinber) [reply src/ts/utils/VolumeController.ts]: As the current calls all happen synchronously, player events will be executed after the whole `finish` function is completed. However, I think there's also no harm in moving it to the end, so fine for me (done in cff68fe4).

**Review by stonko1994 — APPROVED** [review]
